src/passes/Directize.cpp

```diff
@@ -153,7 +153,7 @@ struct FunctionDirectizer : public WalkerPass<PostWalker<FunctionDirectizer>> {
       return replaceWithUnreachable(operands);
     }
     auto* func = getModule()->getFunction(name);
-    if (original->sig != func->getSig()) {
+    if (original->heapType != func->type) {
```

Comment:
Actually, I believe the V8 implementation uses a nominal check here. Performing a structural check on each execution of `call_indirect` would be too slow. I will try to verify this.

Reply:
Do you mean for wasm GC, or in general? I am actually worried about wasm MVP here.

Reply:
If we got rid of the canonicalization in the HeapType constructor then I think this code would become wrong for the MVP? Might be worth at least a comment here, as this line is only correct thanks to that canonicalization (which is in another file far away and also quite surprising, at least to me).

Reply:
Yep, that nominal canonicalization is a gross workaround, I agree. I plan to remove it in a follow-on PR because it seems slightly riskier and will change more behaviors than this change alone. This code won't really break for the MVP, though, since the MVP only has structural function types, not nominal function types. Structural function types are still canonicalized, so `original->heapType == func->type` if and only if those types are structurally equivalent.